### ISSUE TITLE
training log

### ISSUE THREAD

**Issue by sxyysu** [mengmengliu1998/LAformer#14]
Appreciate your work! But during the training process,in the generated train_loss file, FDE: 0.0 MR(2m,4m,6m): (0.0, 0.0, 0.0), is this correct? Does it affect the final evaluation? I look forward to your response and appreciate your assistance. @mengmengliu1998

**Comment by thanh-ntt** [comment]
@sxyysu I am facing the same issue. Can you share how you solved this?

**Comment by sxyysu** [comment]
@thanh-ntt I don't think it matters because it doesn't affect the final result. I obtained similar results by following the README, and the slight differences in results might be attributed to using a single GPU and reducing the batch size.

**Comment by thanh-ntt** [comment]
@sxyysu thank you for the answer!

I also followed the README, but the performance I got is **~10% worse than the results** of the paper. I also compare with the pre-trained models given in the codebase, and my trained model (following README) is also ~10% worse in performance. For example, `MinADE_5 = 1.3207` for `nuscenes_mode_num=5`. Do you mind sharing the configurations / commands that you managed to reproduce the results?

Here's my configuration, btw:
- 2x GPUs, --train_batch_size 512, --num_train_epochs 50 (for both stage_1 and stage_2)
- I use `trainval` split, specifically use flag `--SPLIT train` to pre-process training data, and flag `--SPLIT val` to pre-process evaluation data


Here are the commands that I used for training 2 stages and evaluation, for your reference:

```
# train - stage 1
GPU_NUM=2; \
OUTPUT_DIR=checkpoints/models.laformer.nuscenes.mode_5.topk_2.stage_1; \
TEMP_FILE_DIR=/scratch/e0196773/data/laformer_preprocessed_nuScenes_train_train_meta_v1.0; \
python src/train.py \
--do_train \
--future_frame_num 12 \
--output_dir ${OUTPUT_DIR} \
--topk 2 \
--hidden_size 64 \
--train_batch_size 512 \
--num_train_epochs 50 \
--lane_loss_weight 0.9 \
--distributed_training ${GPU_NUM} \
--temp_file_dir ${TEMP_FILE_DIR} \
--master_port 12356 \
--other_params \
semantic_lane \
direction \
step_lane_score \
enhance_global_graph \
point_level-4-3 \
nuscenes \
nuscenes_mode_num=5
```

```
# train - stage 2
GPU_NUM=2; \
OUTPUT_DIR=checkpoints/models.laformer.nuscenes.mode_5.topk_2.stage_2; \
TEMP_FILE_DIR=/scratch/e0196773/data/laformer_preprocessed_nuScenes_train_train_meta_v1.0; \
MODEL_PATH=/home/svu/e0196773/LAformer/checkpoints/models.laformer.nuscenes.mode_5.topk_2.stage_1/model_save/model.50.bin; \
python src/train.py \
--future_frame_num 12 \
--do_train \
--output_dir ${OUTPUT_DIR} \
--hidden_size 64 \
--topk 2 \
--train_batch_size 512 \
--num_train_epochs 50 \
--lane_loss_weight 0.9 \
--temp_file_dir ${TEMP_FILE_DIR} \
--distributed_training ${GPU_NUM} \
--master_port 12356 \
--other_params \
semantic_lane direction \
step_lane_score \
enhance_global_graph \
point_level-4-3 \
stage-two-train_recover=${MODEL_PATH} \
stage-two-epoch=50 \
stage_two \
nuscenes \
nuscenes_mode_num=5
```

```
# evaluation
MODEL_PATH=checkpoints/models.laformer.nuscenes.mode_5.topk_2.stage_2/model_save/model.50.bin; \
OUTPUT_DIR=checkpoints/laformer.nuscenes.mode_5.topk_2.stage_2; \
GPU_NUM=1; \
TEMP_FILE_DIR=/scratch/e0196773/data/laformer_preprocessed_nuScenes_val_meta_v1.0; \
python src/eval.py \
--future_frame_num 12 \
--eval_batch_size 512 \
--output_dir ${OUTPUT_DIR} \
--hidden_size 64 \
--train_batch_size 32 \
--lane_loss_weight 0.9 \
--topk 2 \
--temp_file_dir ${TEMP_FILE_DIR} \
--model_recover_path ${MODEL_PATH} \
--distributed_training ${GPU_NUM} \
--other_params \
step_lane_score \
stage_two \
semantic_lane \
direction \
enhance_global_graph \
subdivide \
new \
laneGCN \
point_level-4-3 \
stage-two-train_recover=${MODEL_PATH} \
stage-two-epoch=50 \
nuscenes \
nuscenes_mode_num=5 \
--do_eval
```

**Comment by sxyysu** [comment]
@thanh-ntt If you want to reproduce the results in the paper, you can try using 8xRTX3090, and the parameters need to be consistent with the README. I used a single GPU, modified only the batch_size=32 for training and validation, and got MinADE_5 = 1.2306 on the nuScenes dataset for nuscenes_mode_num=5. I think increasing the batch_size should correspond to adjusting the learning rate, rather than using the default value. The learning_rate parameter can be found in LAformer/src/utils_files/config.py.
